contracts/Strategy.sol

```diff
@@ -515,13 +601,15 @@ contract Strategy is BaseStrategy {
             return _amtInWei;
         }
 
-        uint256 price = uint256(chainlinkWantToETHPriceFeed.latestAnswer());
-        return _amtInWei.mul(WAD).div(price);
+        uint256 price = uint256(chainlinkWantUnderlyingTokenToETHPriceFeed.latestAnswer());
+        //TODO: check if this formula is working properly
+        return _amtInWei.mul(EXCHANGE_RATE_PRECISION).div(price).mul(wantAsVault.pricePerShare());
     }
 
-    // ----------------- INTERNAL FUNCTIONS SUPPORT -----------------
-
-    function _repayDebt(uint256 currentRatio) internal {
+    // ----------------- PRIVATE FUNCTIONS SUPPORT -----------------
+    function _repayDebt(uint256 currentRatio) private {
+        //need to compute pending interest
+        abracadabra.accrue();
```

Comment:
@newmickymousse shouldn't interest be computed in `adjustPosition` as well?

Reply:
it should be only in the case we need to repay debt. All the other options (deposit more collateral or mint), don't need to accrue the interests. They will be accrue automatically. In the case of repay debt is to have the real owed value, so we can repay at the accurate value.

Reply:
@newmickymousse hmm what shouldn't you calculate current collateralization ratios with the accrued interest before determining if there is debt to be paid or not?

Reply:
Yes, you are right.